### PR TITLE
Data store migration - Closes #339

### LINKED ISSUE #339: Migrate blockchain data store to new index
### Description

There are two implementations of knex-based library, merge them and make Core service use it.

Add data retention routines.

### Motivation

- Maintainability
- DRY

### Acceptance Criteria

- Index is built properly
- Data retention policy works
- Tests are written and run in CI

### PR DESCRIPTION
**Pull request by MichalTuleja**
### What was the problem?

This PR resolves #339

### How was it solved?

- [x] Kept PoC databases
- [x] Removed PouchDB dependency

### How was it tested?

- Locally

### CODE REVIEW

File: services/core/shared/core/transactionStatistics.js [hunk 1]
```diff
@@ -13,29 +13,40 @@
...
-	if (params.dateFrom || params.dateTo) selector.date = {};
-	if (params.dateFrom) Object.assign(selector.date, { $gte: params.dateFrom.toISOString() });
-	if (params.dateTo) Object.assign(selector.date, { $lte: params.dateTo.toISOString() });
-	// 	WHERE $<dateFrom> <= timestamp AND timestamp <= $<dateTo>
-	result.selector = selector;
-
-	return result;
+	const result = { property: 'date' };
+	if (params.dateFrom) result.from = params.dateFrom.unix();
+	if (params.dateTo) result.from = params.dateTo.unix();
```
Comment (sameersubudhi) [inline services/core/shared/core/transactionStatistics.js:48]: Logical fault!
```suggestion
	if (params.dateTo) result.to = params.dateTo.unix();
```

**Review by sameersubudhi — CHANGES_REQUESTED** [review]
Please find my remarks below:

Additionally, I would be in favour to update the current response structure by removing the `undefined` properties inside of `distributionByType`/`distributionByAmount` and default the `transactionCount` and `volume` properties to `0` instead of `null`.

```json
{
    "data": {
        "timeline": [
            {
                "date": "1970-01-19",
                "transactionCount": null,
                "volume": null,
                "timestamp": 1555200
            }
        ],
        "distributionByType": {
            "undefined": null
        },
        "distributionByAmount": {
            "undefined": null
        }
    },
    "meta": {
        ...
    }
}
```

File: services/core/shared/core/transactionStatistics.js (hunk 1 above)
Comment (MichalTuleja) [reply services/core/shared/core/transactionStatistics.js:48]: Good catch! Thanks :)

**Comment by MichalTuleja** [comment]
@sameersubudhi The development of that feature will continue in #258, and the issues with `undefined` will be fixed there.

**Review by nagdahimanshu — APPROVED** [review]
LGTM!

**Review by sameersubudhi — APPROVED** [review]
Thank you for the update! And, the changes LGTM! 👍

**Pull request merged**